### ISSUE TITLE
Python does not allow multiple entries for Software

### ISSUE THREAD

**Issue by Ichoran** [openworm/tracker-commons#53]
The following minimal file is allowed by the specification but is rejected by the Python implementation

``` json
{ "units":{"t":"s", "x":"mm", "y":"mm"}, "data":[], "metadata":{"software":[{"name":"a"}, {"name":"b"}]} }
```

Since data may undergo multiple processing steps, it is a good idea to support multiple software packages.

**Comment by MichaelCurrie** [comment]
I had missed the comment within the description of the "software" field that "This field may be arrayed".  I will correct this in the schema.

**Comment by MichaelCurrie** [comment]
Fixed by d264380
